program/src/instructions/shared_accounts.rs

```diff
@@ -520,16 +526,33 @@ pub struct MplxShared<'info> {
     pub mint: Box<InterfaceAccount<'info, Mint>>,
 
     /// The Token Metadata metadata account of the NFT.
-    /// CHECK: ownership, structure and mint are checked in assert_decode_metadata.
-    #[account(mut)]
+    /// CHECK: ownership, structure and mint are checked in assert_decode_metadata, seeds checked here.
+    #[account(mut,
+    seeds = [
+        Metadata::PREFIX,
+        mpl_token_metadata::ID.as_ref(),
+        mint.key().as_ref(),
+    ],
+    bump,
+    seeds::program = mpl_token_metadata::ID,
+    )]
     pub metadata: UncheckedAccount<'info>,
 
-    // --------------------------------------- pNft
     /// The Token Metadata edition account of the NFT.
-    /// CHECK: seeds checked on Token Metadata CPI
-    //note that MASTER EDITION and EDITION share the same seeds, and so it's valid to check them here
+    /// CHECK: seeds checked here
+    #[account(
+        seeds = [
+            MasterEdition::PREFIX.0,
+            mpl_token_metadata::ID.as_ref(),
+            mint.key().as_ref(),
+            MasterEdition::PREFIX.1,
```

Comment:
What if they're not master editions?

Reply:
Edition derivations are the same derivation as MasterEditions.